tests/test_script_helper.py

```diff
@@ -225,318 +287,312 @@ def generate_random_synapses_locations(cell,syn_filename,num_syns=10000,seed=2):
+                     ICLAMP_DURATION=ICLAMP_PARAMS['duration'],
+                     TSTOP=SIM_PARAMS['tstop'],
+                     DT=SIM_PARAMS['dt'],
+                     SPIKE_BEGIN_TIME=SPIKE_BEGIN_TIME,
+                 ))
+        for i in range(int(recording_vec_control.size())):
+            fd.write(repr(recording_vec_control.x[i]) + "\n")
+        for i in range(int(recording_vec_reduced.size())):
+            fd.write(repr(recording_vec_reduced.x[i]) + "\n")
+
+
+def load_voltages(voltage_file, number_of_recordings):
+    '''load_voltages'''
+    with open(voltage_file) as fd:
+        fd.readline()   # skips first line with the simulation configuration details
+        voltages = [float(v) for v in fd]
+
+    assert len(voltages) == number_of_recordings * 2
+    np_orig_recording_vec_control = np.array(voltages[:number_of_recordings])
+    np_orig_recording_vec_reduced = np.array(voltages[number_of_recordings:])
+    return np_orig_recording_vec_control, np_orig_recording_vec_reduced
+
+
+def setup_iclamps(sec, params):
+    '''sets iclamp for `sec` with `params`'''
+    iclamp_reduced_cell = h.IClamp(0.5, sec=sec)
+    iclamp_reduced_cell.delay = params['delay']
+    iclamp_reduced_cell.dur = params['duration']
+    iclamp_reduced_cell.amp = params['amplitude']
+    return iclamp_reduced_cell
+
+
+def run_test(morphology_file,
+             model_file,
+             model_file_reduced,
+             reduction_frequency,
+             manual_total_nsegs,
+             synapse_file,
+             voltage_file,
+             write_unit_test_vectors,
+             plot_voltages,
+             create_type,
+             celsius):
+    '''Run a test reduction
+
+    Note: This must be run from a new python instance so that neuron is reset
     '''
-    global REDUCTION_FREQUENCY, MODEL_FILE, SYNAPSE_FILE, MORPHOLOGY_FILE, MANUAL_TOTAL_NSEGS
-    
-    #First Delete all sections
-    h("forall delete_section()")
-    h.celsius = celsius
-    
-    #Set variables
-    REDUCTION_FREQUENCY = int(frequency)
-    MODEL_FILE = orig_model_file
-    MODEL_FILE_REDUCED = reduced_model_file
-    SYNAPSE_FILE = synapse_file
-    MORPHOLOGY_FILE = orig_morphology_file
-    MANUAL_TOTAL_NSEGS = manual_total_nsegs
-    VOLTAGEFILE = voltage_file
+    h.celsius = celsius  # needs to be set for reduction to work properly
+
+    reduction_frequency = int(reduction_frequency)
     print('---------------sim data --------------------------')
-    print("REDUCTION_FREQUENCY " + str(REDUCTION_FREQUENCY))
-    print("MODEL_FILE          " + str(MODEL_FILE))
-    print("MODEL_FILE_REDUCED  " + str(MODEL_FILE_REDUCED))
-    print("SYNAPSE_FILE        " + str(SYNAPSE_FILE))
-    print("MORPHOLOGY_FILE     "  + str(MORPHOLOGY_FILE))
-    print("MANUAL_TOTAL_NSEGS  " + str(MANUAL_TOTAL_NSEGS))
-    print("VOLTAGEFILE         "  + str(VOLTAGEFILE))
+    print("REDUCTION_FREQUENCY %s" % reduction_frequency)
```

Comment:
Why you do not use format?

Reply:
Old habits die hard: I basically use `%` when there are only a couple of arguments, I want something automatically converted to a string (so I don't have to use `str()`). `.format()` requires more typing as well, but it's useful for named replacement